### PR TITLE
Update statsd to new adapter form

### PR DESCRIPTION
**Pull request by ZackButcher**
```release-note
NONE
```

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/1136)

### CODE REVIEW

File: adapter/statsd2/statsd.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+var (
+	_ metric.HandlerBuilder = &builder{}
+	_ metric.Handler        = &handler{}
+)
+
+// GetBuilderInfo returns the BuilderInfo associated with this adapter implementation.
+func GetBuilderInfo() adapter.BuilderInfo {
+	return adapter.BuilderInfo{
+		Name:        "istio.io/mixer/adapter/statsd2",
```
Comment (geeknoid) [inline adapter/statsd2/statsd.go]: Can just put statds so we don't forget when doing the file-level rename.

File: adapter/statsd2/statsd.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+		if !found {
+			env.Logger().Infof("template registered for nonexistent metric '%s'", metricName)
+			continue // we don't have a metric that corresponds to this template, skip processing it
+		}
+
+		var t *template.Template
+		if s.NameTemplate != "" {
+			t, _ = template.New(metricName).Parse(s.NameTemplate)
+			if err := t.Execute(ioutil.Discard, def.Dimensions); err != nil {
+				env.Logger().Warningf(
```
Comment (geeknoid) [inline adapter/statsd2/statsd.go]: Having mistyped stuff in the template seems like a very common/likely problem. Can we catch this in validateConfig instead so the user gets the error at the right place?

File: adapter/statsd2/config/config.proto [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+option (gogoproto.gostring_all) = false;
+
+message Params {
+    // Address of the statsd server, e.g. localhost:8125
+    string address = 1;
+
+    // Metric prefix, do not specify for no prefix
+    string prefix = 2;
+
+    // Flush Interval controls the maximum amount of time between sending metrics to the statsd collection server.
```
[the remark this answers is not in the document]
Comment (ZackButcher) [reply adapter/statsd2/config/config.proto]: Done

File: adapter/statsd2/config/config.proto [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    // Metrics are reported when either flush_bytes is full or flush_interval time has elapsed since the last report.
```
[the remark this answers is not in the document]
Comment (ZackButcher) [reply adapter/statsd2/config/config.proto]: Done

File: adapter/statsd2/statsd.go (hunk 2 above)
Comment (ZackButcher) [reply adapter/statsd2/statsd.go]: We do already - lines 90-94. We can remove the check here entirely, but that felt a little off.

**Comment by ZackButcher** [comment]
PTAL

**Pull request merged**